Home4

IgorPrihodko opened the pull request.

IgorPrihodko committed:
Homework_1

IgorPrihodko committed:
Homework_2

IgorPrihodko committed:
Homework_3

IgorPrihodko committed:
Homework_4

Sofasmile commented on src/main/java/homework4/task1/Robot.java:
```diff
@@ -0,0 +1,61 @@
+public class Robot {
+
+    public Direction getDirection() {
+       return this.getDirection(); // текущее направление взгляда
```
better delete the comments
If you want write them, you must use English

Sofasmile commented on src/main/java/homework4/task1/Robot.java:
```diff
@@ -0,0 +1,61 @@
+        // за один шаг робот изменяет одну свою координату на единицу
+    }
+
+    public enum Direction {
```
better move enum in the separate file

Sofasmile commented on src/main/java/homework4/task1/Robot.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    public void turnLeft() {
+        // повернуться на 90 градусов против часовой стрелки
```
write the implementation of this method
it will be useful for you: more practice - more experience

Sofasmile commented on src/main/java/homework4/task1/Robot.java:
```diff
@@ -0,0 +1,61 @@
+
+    public static void moveRobot(Robot robot, int toX, int toY) {
+        if (robot.getX() > toX) {
+            while (robot.getDirection() != Direction.LEFT)
```
CS
use { }

Sofasmile commented on src/main/java/homework4/task2/ComplexNumber.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
```
CS

Sofasmile commented on src/main/java/homework4/task3/AsciiCharSequence.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    public char charAt(int index) {
+        return (char)(mass[index] & 0xff);
```
do you need (& 0xff) here?
if yes, write why. And move it into separate static final variable in the class

Sofasmile commented on src/main/java/homework4/task4/CheckLabelsMethod.java:
```diff
@@ -0,0 +1,13 @@
+package homework4.task4;
+
+public class CheckLabelsMethod {
```
why you write Method in the class name?

Sofasmile commented on src/main/java/homework4/task4/KeywordAnalyzer.java:
```diff
@@ -0,0 +1,18 @@
+    protected abstract Label getLabel();
+
+    public Label processText(String text) {
+
```
cs

Sofasmile commented on src/main/java/homework4/task4/NegativeTextAnalyzer.java:
```diff
@@ -0,0 +1,21 @@
+
+    private String[] keywords = {":(", "=(", ":|"};
+    private Label label = Label.NEGATIVE_TEXT;
+    @Override
```
cs

Sofasmile commented on src/main/java/homework4/task5/MathSqrtException.java:
```diff
@@ -0,0 +1,13 @@
+
+        if (x < 0) {
+            throw new IllegalArgumentException("Expected non-negative number, got " + x);
+        } else {
```
delete else statement
it will work without it

Sofasmile reviewed (commented):
create another example of builder
read about it
https://habr.com/ru/post/244521/

IgorPrihodko committed:
Homework_4 Fixed Builder. Created own file for enum, deleted cooments,
added realization of turnLeft method. Fixed CS problems. Fixed charAt
method. Changed name of class CheckLabels.Deleted else statement in
MathSqrtException class.

Sofasmile reviewed (approved):
good
